File: crop_logos.js
const fs = require('fs');
const path = require('path');
const sharp = require('sharp');
const mongoose = require('mongoose');

const imgPath = '/Users/sushant/.gemini/antigravity/brain/f879f993-65c5-4d74-bc62-0de2c466c2c9/.user_uploaded/media__1785714552059.png';
const fallbackImgPath = '/Users/sushant/.gemini/antigravity/brain/f879f993-65c5-4d74-bc62-0de2c466c2c9/.user_uploaded/media__1785679944644.png';
const outDir = path.join(__dirname, 'uploads', 'team_logos');

const teams = [
    [{ name: "Janakpur Bolts", short: "jb" }, { name: "Sudurpaschim Royals", short: "sr" }],
    [{ name: "Karnali Yaks", short: "ky" }, { name: "Kathmandu Gurkhas", short: "kg" }],
    [{ name: "Pokhara Avengers", short: "pa" }, { name: "Lumbini Lions", short: "ll" }],
    [{ name: "Chitwan Rhinos", short: "cr" }, { name: "Biratnagar Kings", short: "bk" }]
];

async function run() {
    if (!fs.existsSync(outDir)) fs.mkdirSync(outDir, { recursive: true });

    let activeImgPath = imgPath;
    if (!fs.existsSync(activeImgPath)) {
        activeImgPath = fallbackImgPath;
    }
    
    if (!fs.existsSync(activeImgPath)) {
        console.log("No valid image found.");
        process.exit(1);
    }

    const image = sharp(activeImgPath);
    const metadata = await image.metadata();
    
    console.log(`Image loaded: ${metadata.width}x${metadata.height}`);
    
    const cellW = Math.floor(metadata.width / 2);
    const cellH = Math.floor(metadata.height / 4);

    require('dotenv').config();
    await mongoose.connect(process.env.MONGO_URI || 'mongodb://127.0.0.1:27017/smart-stadium');
    const Team = mongoose.connection.collection('teams');

    for (let row = 0; row < 4; row++) {
        for (let col = 0; col < 2; col++) {
            const team = teams[row][col];
            const left = col * cellW;
            const top = row * cellH;
            
            const savePath = path.join(outDir, `${team.short}.png`);
            
            await image.clone()
                .extract({ left, top, width: cellW, height: cellH })
                .toFile(savePath);
                
            const logoUrl = `/uploads/team_logos/${team.short}.png`;
            const res = await Team.updateOne({ name: team.name }, { $set: { logoUrl } });
            
            console.log(`Saved ${team.name} to ${savePath}, DB modified: ${res.modifiedCount}`);
        }
    }
    
    console.log("Done!");
    process.exit(0);
}

run().catch(console.error);

File: crop_logos.py
import os
from PIL import Image
from pymongo import MongoClient
import sys

img_path = '/Users/sushant/.gemini/antigravity/brain/f879f993-65c5-4d74-bc62-0de2c466c2c9/.user_uploaded/media__1785714552059.png'
if not os.path.exists(img_path):
    print("File not found.")
    sys.exit(1)

img = Image.open(img_path)
width, height = img.size
print(f"Image loaded: {width}x{height}")

# If it's a phone screenshot, it's tall. If it's the 2x4 grid, it's roughly square or portrait but grid-like.
# A 2x4 grid of logos is usually roughly 1:1 or 1:2.

cell_w = width // 2
cell_h = height // 4

teams = [
    [("Janakpur Bolts", "jb"), ("Sudurpaschim Royals", "sr")],
    [("Karnali Yaks", "ky"), ("Kathmandu Gurkhas", "kg")],
    [("Pokhara Avengers", "pa"), ("Lumbini Lions", "ll")],
    [("Chitwan Rhinos", "cr"), ("Biratnagar Kings", "bk")]
]

out_dir = '/Users/sushant/Desktop/smart-stadium/backend/uploads/team_logos'
os.makedirs(out_dir, exist_ok=True)

client = MongoClient('mongodb://127.0.0.1:27017/smart-stadium')
db = client.get_database()
teams_col = db.teams

for row in range(4):
    for col in range(2):
        team_name, short = teams[row][col]
        left = col * cell_w
        upper = row * cell_h
        right = left + cell_w
        lower = upper + cell_h
        
        box = (left, upper, right, lower)
        cropped = img.crop(box)
        
        save_path = os.path.join(out_dir, f"{short}.png")
        cropped.save(save_path)
        
        # Update MongoDB
        logo_url = f"/uploads/team_logos/{short}.png"
        res = teams_col.update_one({'name': team_name}, {'$set': {'logoUrl': logo_url}})
        print(f"Saved {team_name} to {save_path}, DB modified: {res.modified_count}")

print("Done!")
